### PR TITLE
T/590 Replace old dialog handler with a public method for selecting color

### LINKED ISSUE #590: Colorbutton plugin should utilize official API for getting color from dialog
## Are you reporting a feature or a bug?

Task

## Description

[Color Dialog](http://ckeditor.com/addon/colordialog) plugin exposes [public method for getting selected color, `getColorFromDialog`](http://docs.ckeditor.com/#!/api/CKEDITOR.editor-method-getColorFromDialog). However [Color Button](http://ckeditor.com/addon/colorbutton) plugin, the main user of Color Dialog, is not using it internally. It reinvents it with [its own dialog handler](https://github.com/ckeditor/ckeditor-dev/blob/master/plugins/colorbutton/plugin.js#L205-L248). It would be nice to reuse official API here.

## Other details

* Installed CKEditor plugins: colorbutton, colordialog

### PR DESCRIPTION
**Pull request by beatadelura**
## What is the purpose of this pull request?

Task

## Does your PR contain necessary tests?
Yes. ⚠️  Please check whether this is enough.

### This PR contains

- [X] Unit tests
- [X] Manual tests

## What changes did you make?

Replace old dialog handler with a public method for selecting color.

---

Closes #590

### CODE REVIEW

File: tests/plugins/colorbutton/colorbutton.js [hunk 1]
```diff
@@ -81,6 +81,37 @@
 				bgColorBtn.click( editor );
 				assert.areEqual( 0, bgColorBtn._.panel.getBlock( bgColorBtn._.id ).element.find( '.cke_colorauto' ).count(), 'Automatic button should not be visible.' );
 			} );
+		},
+
+		// 590
+		'test changing text color': function() {
```
Comment (Comandeer) [inline tests/plugins/colorbutton/colorbutton.js]: I'm lacking unit test for second scenario, which is described in manual test.

File: plugins/colorbutton/plugin.js [hunk 2]
```diff
@@ -241,10 +232,10 @@ CKEDITOR.plugins.add( 'colorbutton', {
 						return !( element.is( 'a' ) || element.getElementsByTag( 'a' ).count() ) || isUnstylable( element );
 					};
 
+					editor.focus();
```
Comment (Comandeer) [inline plugins/colorbutton/plugin.js]: Shouldn't snapshot be saved after applying color?

File: tests/plugins/colorbutton/colorbutton.js (hunk 1 above)
Comment (beatadelura) [inline tests/plugins/colorbutton/colorbutton.js]: Done 👍

File: plugins/colorbutton/plugin.js (hunk 2 above)
Comment (beatadelura) [inline plugins/colorbutton/plugin.js]: 👍

File: tests/plugins/colorbutton/colorbutton.js [hunk 3]
```diff
@@ -81,6 +101,48 @@
...
+			onDialogShow( ed );
+			bot.setHtmlWithSelection( '[<h1>Moo</h1>]' );
+
+			txtColorBtn.click( ed );
+			openColorDialog( frame );
+		},
+
+		// 590
+		'test changing text color with no focus': function() {
+			var editor = CKEDITOR.replace( 'editor' ),
```
Comment (Comandeer) [inline tests/plugins/colorbutton/colorbutton.js]: You could use `bender.editorBot.create` to create new editor (see other tests in this file).

File: tests/plugins/colorbutton/colorbutton.js [hunk 4]
```diff
@@ -81,6 +101,48 @@
...
+					} );
+
+					onDialogShow( editor );
+
+					txtColorBtn.click( editor );
+					openColorDialog( frame );
+				} );
+			} );
+
+			wait();
```
Comment (Comandeer) [inline tests/plugins/colorbutton/colorbutton.js]: Actually I'm missing checking if undo/redo is working correctly in both of unit tests.

File: tests/plugins/colorbutton/colorbutton.js (hunk 3 above)
Comment (beatadelura) [inline tests/plugins/colorbutton/colorbutton.js]: Done 👍

File: tests/plugins/colorbutton/colorbutton.js (hunk 4 above)
Comment (beatadelura) [inline tests/plugins/colorbutton/colorbutton.js]: @Comandeer I wrote unit tests for undo/redo, PTAL 👀

**Pull request merged**